cli/pcluster/easyconfig.py

```diff
@@ -63,32 +64,69 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
-def prompt(message, default_value=None, hidden=False, options=None, check_validity=False):
-    if hidden and default_value is not None:
-        user_prompt = message + " [*******" + default_value[-4:] + "]: "
-    else:
-        user_prompt = message + " ["
-        if default_value is not None:
-            user_prompt = user_prompt + default_value + "]: "
-        else:
-            user_prompt = user_prompt + "]: "
+def prompt(message, default_value=None, options=None):
+    """
+    Prompt the user a message and return the acquired value.
+
+    Example:
+
+    prompt("vpc", "id1", [("id1", "sasso"), ("id2", "cavolo")])
 
-    if isinstance(options, (list, tuple)):
+    result:
+    __________________
+    Acceptable value for vpc:
+    1) id1 | sasso
+    2) id2 | cavolo
+
+    vpc[id1] = xxx
+    __________________
+
+    :param message: the message to show to the user
+    :param default_value: the value to return if user does not input anything
+    :param options: a list of tuple (can also be list/list or tuple/tuple) containing the options to show to the user.
+    Only the first element of the tuple will be considered as acceptable.
+    The user can then choose one of the option or write something that is not in them.
+    :raise ValueError if options is not a list of tuple
+    :return: the acquired value.
+    """
+    user_prompt = "{0} [{1}]: ".format(message, default_value or "")
+    if options:
+        if not (isinstance(options, (tuple, list))):
+            # Should never been thrown, DEBUGGING
+            raise ValueError("options is not a list nor a tuple")
+        if not (isinstance(options[0], (tuple, list))):
+            # Convert a list to a list of tuple
+            options = [(item,) for item in options]
         print("Acceptable Values for %s: " % message)
-        for o in options:
-            print("    %s" % o)
+        allowed_values = set([item[0] for item in options])
+        numbered_items = NumberedList(options)
+        for key, value in numbered_items.get().items():
+            print("{0}. {1}".format(key, " | ".join(value)))
+
+        result = get_correct_answer(allowed_values, default_value, message, numbered_items, user_prompt)
+    else:
+        user_input = input(user_prompt).strip()
+        result = user_input if user_input != "" else default_value
```

Comment:
you could write:
` result = user_input if user_input else default_value`
or
` result = user_input or default_value`